### DISCUSSION TITLE
Support for T-Series possible?

### DISCUSSION THREAD

**Discussion by nikowitt** [nathanmarlor/foxess_modbus#266] [Q&A]
Hi there,

I am using a Foxess T-13 inverter. Is it possible to add support for this inverter? My idea would be to connect Modbus via USB adapter. There is a repository https://github.com/assembly12/Foxess-T-series-ESPHome-Home-Assistant that provides support, but uses an ESP based solution which I would like to avoid.

Do you see any change realizing this? Is there is any possibility for me to contribute?

Regards,
Niko

**Comment by canton7** [comment]
Reading through the project you linked, it looks like the T-series works in a different way to the H-series. 

The H-series expose a number of registers over modbus. Each register holds a different sort of value (PV1 voltage, etc), and we query individual registers using the modbus protocol.

The T-series doesn't seem to expose a modbus interface, but instead just transmits a set of values every few minutes. 

It should be possible to use an off-the-shelf device RS485 to USB converter, or a device which receives packets over RS485 and re-transmits them over TCP (the W610 looks like it might be able to do this, but the documentation is rubbish), which would give you a route to avoiding ESPHome.

However as things stand, this integration isn't really set up to do that: it would need a fair amount of rework, and to be honest it would probably be easier to start again with a new integration. That's going to be much easier for someone who has a T-series inverter to develop and test against: doing it entirely removely will be a bit painful.

So, I'm afraid the straight answer is probably "no". But, with a bit of work it should be possible to use an off-the-shelf RS485 to USB converter, however you will need to invest some time into writing a new integration.

**Comment by nikowitt** [comment]
Thanks for your response. I will have a look at this once the hardware to attach to the inverter has arrived.
